crates/bevy_ecs/src/schedule/state.rs

```diff
@@ -64,6 +82,24 @@ impl StateCallback {
     }
 }
 
+pub struct StateTransitionConfig<'a, T>
+where
+    T: Component + Debug + Clone + Eq + Hash,
+{
+    state: &'a mut State<T>,
+}
+
+impl<'a, T> StateTransitionConfig<'a, T>
+where
+    T: Component + Debug + Clone + Eq + Hash,
+{
+    /// Controls where the transition will land, by default in the same frame.
+    pub fn next(self, next: Next) -> Self {
```

Comment:
```suggestion
    pub fn next(self, next: Next) {
```
No need to return `Self` since we only have the one function for the config. (we can change/extend this later if necessary).

-----
Are `next` and `Next` the best names for this?
perhaps something like `on_frame` and `OnFrame` perhaps?

@mockersf , sorry for the slew of comments haha, however this _should_ be my last one :wink:

Reply:
> Are `next` and `Next` the best names for this?

No! And I dislike the signature with that many `Next`... not a fan of `OnFrame` either...
Actually the function exposed just needs to be `end_frame()` as you only need it to net do the default

> sorry for the slew of comments

No problem! please comment if you don't like my latest changes 👍